src/hotspot/share/utilities/rbTree.inline.hpp

```diff
@@ -479,9 +481,9 @@ inline void RBTree<K, V, COMPARATOR, ALLOCATOR>::remove(RBNode* node) {
 template <typename K, typename V, typename COMPARATOR, typename ALLOCATOR>
 template <typename F>
 inline void RBTree<K, V, COMPARATOR, ALLOCATOR>::visit_in_order(F f) const {
-  RBNode* to_visit[64];
+  const RBNode* to_visit[64];
```

Comment:
Note to self or others: This needs at least an assertion, better a visit-stop on release builds

Reply:
An assertion is nice-to-have. A "visit-stop", you mean we should break if we exceed the limit?

I was thinking about this in the original PR. If we do actually have a RB-tree then we're bounded to 2log(n) depth, so this should be impossible to reach. That is, if we do have a RB-Tree, a bug would invalidate that assumption.

Reply:
Our linters will definitely complain about that suspected buffer override and are probably not smart enough to figure out the tree limit. Definitely an assert. As long as we don't use the iteration for anything too intensive, either a guarantee or a bailout. If we start using iteration in performance-critical paths, we can rethink.